### ISSUE TITLE
Hyperparameter tuning

### ISSUE THREAD

**Issue by cbarkr** [shabbiryusufali/vole#38]
# Background
- Tuned hyperparameters can make or break a model
- We don't need to tune them manually - tools like [Optuna](https://optuna.org/) do this for us

# Deliverables
- Integrate [`optuna`](https://pypi.org/project/optuna/) into `train.py` to tune `hidden_channels`, `num_layers`, `dropout`, and `lr` (note: see https://www.geeksforgeeks.org/deep-learning/hyperparameter-tuning-with-optuna-in-pytorch/ for a tutorial)
- Potentially even let Optuna choose the optimizer itself! (note: see https://readmedium.com/tuning-pytorch-hyperparameters-with-optuna-470edcfd4dc)
- Adjust `save_model` such that we can recover `hidden_channels` and `num_layers` (note: will need this to be able to load the model later!)
